parser/parser.go

```diff
@@ -76,42 +79,48 @@ func buildEC2ARN(trimmedGroup, accountID, region string) string {
 	// expect vpc/{vpc_id} or instance/{instance_id}
 	parts := strings.Split(trimmedGroup, "/")
 	if len(parts) == 2 {
-		return BuildServiceARN("ec2", accountID, region, fmt.Sprintf("%s/%s", parts[0], parts[1]))
+		return BuildResourceARN("ec2", accountID, region, fmt.Sprintf("%s/%s", parts[0], parts[1]))
 	}
 
 	// as fallback assume logs are from an ec2 instance and use ec2/{instance_id}
-	return BuildServiceARN("ec2", accountID, region, fmt.Sprintf("instance/%s", trimmedGroup))
+	return BuildResourceARN("ec2", accountID, region, fmt.Sprintf("instance/%s", trimmedGroup))
 }
 
-func buildECSARNs(trimmedGroup, accountID, region string) []string {
-	parts := strings.Split(trimmedGroup, "/")
+func buildECSARNs(trimmedGroup, logStream, accountID, region string) []string {
+	groupParts := strings.Split(trimmedGroup, "/")
+	streamParts := strings.Split(logStream, "/")
 
-	if len(parts) == 1 {
-		return []string{BuildServiceARN("ecs", accountID, region, fmt.Sprintf("cluster/%s", parts[0]))}
+	arns := make([]string, 0)
+	if len(streamParts) == 3 {
+		arns = append(arns, BuildResourceARN("ecs", accountID, region, fmt.Sprintf("task/%s/%s", groupParts[0], streamParts[2])))
 	}
 
-	if len(parts) == 2 {
-		return []string{
-			BuildServiceARN("ecs", accountID, region, fmt.Sprintf("cluster/%s", parts[0])),
-			BuildServiceARN("ecs", accountID, region, fmt.Sprintf("service/%s/%s", parts[0], parts[1])),
-		}
+	if len(groupParts) == 1 {
+		arns = append(arns, BuildResourceARN("ecs", accountID, region, fmt.Sprintf("cluster/%s", groupParts[0])))
+	} else if len(groupParts) == 2 {
+		arns = append(arns, BuildResourceARN("ecs", accountID, region, fmt.Sprintf("cluster/%s", groupParts[0])))
+		arns = append(arns, BuildResourceARN("ecs", accountID, region, fmt.Sprintf("service/%s/%s", groupParts[0], groupParts[1])))
+
+	} else {
+		// as fallback ecs/{resource_name}
+		arns = append(arns, BuildResourceARN("ecs", accountID, region, trimmedGroup))
```

Comment:
using `switch` in this like cases make code more readable, and seems like `len(groupParts) == 1` and `len(groupParts)==2` are same, you can combine them into same case statement, and final else can be default.